src/api/app/components/diff_list_component.html.haml

```diff
@@ -4,24 +4,23 @@
     old_filename = file_info.dig('old', 'name')
     expanded = expand?(name, state, file_index)
   .accordion.mb-2.diff-accordion{ id: "diff-list-#{name.parameterize}" }
-    - if file_info["diff"]["binary"] == "1"
-      .card.p-3.mb-2
-        %span
-          = render(DiffSubjectComponent.new(state:, old_filename:, new_filename: name))
-    - else
-      .accordion-item
-        %h2.accordion-header
+    .accordion-item
+      %h2.accordion-header
+        - if file_info['disabled']
```

Comment:
```suggestion
        - if file_info.dig('disabled')
```

Reply:
What is the purpose of this `dig`?
```
[1] pry(main)> nil['disabled']
NoMethodError: undefined method '[]' for nil (NoMethodError)
from (pry):1:in '__pry__'
[2] pry(main)> nil.dig('disabled')
NoMethodError: undefined method 'dig' for nil (NoMethodError)
from (pry):2:in '__pry__'
```

Reply:
AFAIR `file_info` can be nil right?

Reply:
But you will have to handle this somehow right? :)

Reply:
Well, now that I look into it, it won't be nil, because if it is nil it would crash upstream (for example [here](https://github.com/openSUSE/open-build-service/pull/18196/files#diff-bc13fae0e87d70e18ab81ed73195697a7c3cad5a62b2301ea2c82ad37f2375fbR45)) and this is filled in by [webui_sourcediff](https://github.com/openSUSE/open-build-service/blob/642ff8dd20a5b517e5b607b239607391ac7d6943/src/api/app/models/bs_request_action.rb#L258)

Reply:
The [nil error](https://github.com/openSUSE/open-build-service/pull/18312) we had was because of trying to get nested values from a hash. Now it’s not a nested value. It should be fine

Reply:
Then how was it nil last time?

https://github.com/openSUSE/open-build-service/pull/18312

Reply:
Because `file_info['diff']` was nil, then `file_info['diff']['binary']` was a certain crash. In this case, with the `dig` in place, the only other nil that could happen is if `file_info` could be nil. And this only can be nil if we somehow end up asking for a file that we don't have in the hash created [here](https://github.com/openSUSE/open-build-service/blob/e510b8b5db2311726aeba2d5bd240e16df4eaa11/src/api/app/mixins/parse_package_diff.rb#L18)